exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/InsertStatement.kt

```diff
@@ -69,7 +69,23 @@ open class InsertStatement<Key : Any>(
             if (firstAutoIncColumn != null || returnedColumns.isNotEmpty()) {
                 while (rs?.next() == true) {
                     val returnedValues = returnedColumns.associateTo(mutableMapOf()) { it.first to rs.getObject(it.second) }
-                    if (returnedValues.isEmpty() && firstAutoIncColumn != null) returnedValues[firstAutoIncColumn] = rs.getObject(1)
+                    if (returnedValues.isEmpty() && firstAutoIncColumn != null) {
+                        try {
+                            returnedValues[firstAutoIncColumn] = rs.getObject(1)
+                        } catch (e: ArrayIndexOutOfBoundsException) {
+                            // EXPOSED-191 Flaky Oracle test on TC build
+                            // this try/catch should help to get information about the flaky test.
+                            // try/catch can be safely removed after the fixing the issue.
+                            // TooGenericExceptionCaught suppress also can be removed
+
+                            exposedLogger.error(
+                                "ArrayIndexOutOfBoundsException on processResults. " +
+                                    "Table: ${table.tableName}, firstAutoIncColumn: ${firstAutoIncColumn.name}, inserted: $inserted",
+                                e
+                            )
```

Comment:
Could the culprit statement itself also be logged? `this.prepareSQL(TransactionManager.current(), prepared = true)`
Since there seem to be many reports of this exception with prepared statements and the amount of parameters (or maybe the set fetch size).

Another option could be to enable debug mode in the test's transaction and log the statements as they are executed there, but because the test uses DAO, the `commit()` wouldn't allow us to single out which statement is failing.

And if the logs don't give us enough info, I'd question why `commit()` is even being used.
The SELECT COUNT should trigger a full flush before all the asserts are run and the test should pass without it.

Reply:
Thank you, I didn't know about the `TransactionManager.current()` options)

Added logging of sql that failed and logging of columns that should taken from the result set.

Also added stdout logger to the test.